modules/integration/tests-common/integration-test-utils/src/main/java/ues/integration/tests/common/domain/UESIntegrationTest.java

```diff
@@ -0,0 +1,118 @@
+package ues.integration.tests.common.domain;
+
+/*
+*Copyright (c) 2015​, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*WSO2 Inc. licenses this file to you under the Apache License,
+*Version 2.0 (the "License"); you may not use this file except
+*in compliance with the License.
+*You may obtain a copy of the License at
+*
+*http://www.apache.org/licenses/LICENSE-2.0
+*
+*Unless required by applicable law or agreed to in writing,
+*software distributed under the License is distributed on an
+*"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*KIND, either express or implied.  See the License for the
+*specific language governing permissions and limitations
+*under the License.
+*/
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.testng.Assert;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.context.beans.Tenant;
+import org.wso2.carbon.automation.engine.context.beans.User;
+import org.wso2.carbon.automation.test.utils.common.TestConfigurationProvider;
+
+import javax.xml.xpath.XPathExpressionException;
+
+public abstract class UESIntegrationTest {
+    private static final Log LOG = LogFactory.getLog(UESIntegrationTest.class);
+    protected final TestUserMode userMode;
+    private User userInfo = null;
+    private AutomationContext uesContext = null;
+    private Tenant tenantInfo = null;
+
+
+    public UESIntegrationTest(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    public UESIntegrationTest() {
+        this(TestUserMode.SUPER_TENANT_ADMIN);
+    }
+
+    public AutomationContext getUesContext() throws XPathExpressionException {
```

Comment:
Hi Manu,
I have bit confusion on this why we are making this as private rather than protected or public.I have referred http://stackoverflow.com/questions/21016776/how-can-a-derived-class-invoke-private-method-of-base-class too.Can you give me more idea on this?